### PR TITLE
Add jetson prediction pipeline

### PR DESCRIPTION
**Pull request by jonathan-paugh**
Adds a prediction pipeline that allows the jetson nano to take a snapshot an make a prediction based on a trained model.

Additional changes:
- Optimize hardware settings when on jetson nano
- Move model training into a function for reusability
- Refactor pickling logic to restore training metrics (history object) so they can be used when importing model as well
- Added `load_image_tensor(...)` function to import image files without requiring a directory
  - Tensorflow uses a different pixel format so be sure to use this function instead of any other when reading image files from disk
- Changed model classification output to 'softmax'
- Add `OptionInput` class to library to allow for menu functionality

⚠️ Refactored the dataset directory to hold files in dataset/train and dataset/test for training and testing respectively. Please update your local directories to mirror this.

### CODE REVIEW

File: configs/model.py [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-IMAGE_SIZE = (512, 384)
+IMAGE_SIZE = (96, 72)
 
 MODEL_NUM_EPOCHS = 5
-MODEL_BATCH_SIZE = 1024
-MODEL_WORKERS = 32
+MODEL_WORKERS = os.cpu_count() | 1
+MODEL_BATCH_SIZE = int(math.pow(2, 3)) if IS_JETSON else int(math.pow(2, 6))
+
+LOADER_BATCH_SIZE = int(math.pow(2, 3)) if IS_JETSON else int(math.pow(2, 5))
```
Comment (semibran) [inline configs/model.py]: use `x ** y` syntax for these

File: core/loader.py [hunk 2]
```diff
@@ -1,17 +1,53 @@
...
+def load_image_tensor(path, image_size=config.IMAGE_SIZE, color_mode='rgb', interpolation='bilinear'):
+    if color_mode == 'rgb':
+        num_channels = 3
+    elif color_mode == 'rgba':
+        num_channels = 4
+    elif color_mode == 'grayscale':
+        num_channels = 1
+    else:
+        raise ValueError(
+            '`color_mode` must be one of {"rbg", "rgba", "grayscale"}. '
```
Comment (semibran) [inline core/loader.py]:
```suggestion
            '`color_mode` must be one of {"rgb", "rgba", "grayscale"}. '
```

**Review by semibran — APPROVED** [review]
